contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_v4l2_video_decoder_performance.py

```diff
@@ -0,0 +1,237 @@
+        type=str,
+        help="Path of Golden Sample file",
+    )
+
+    parser.add_argument(
+        "-dp",
+        "--decoder_plugin",
+        required=True,
+        type=str,
+        help="Decoder plugin be used in gstreamer pipeline e.g. v4l2h264dec",
+    )
+
+    parser.add_argument(
+        "-s",
+        "--sink",
+        default="fakesink",
+        type=str,
+        help=("Specific sink that helps on judgement (default: fakesink)"),
+    )
+
+    parser.add_argument(
+        "-mf",
+        "--minimum_fps",
+        required=True,
+        type=str,
+        help=(
+            "The minimum value of FPS that "
+            "all average FPS value should not violate"
+        ),
+    )
+
+    parser.add_argument(
+        "-pmt",
+        "--performance_mode_target",
+        default="",
+        type=str,
+        help=(
+            "Once this value be assigned, script will enable performance mode"
+            " by the given value. For instance, if 'genio-1200' be assigned, "
+            "script will find the performance controller and execute the logic"
+            ' of genio-1200 board. (Default: "", will do nothing)'
+        ),
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def build_gst_command(
+    gst_bin: str, golden_sample_path: str, decoder: str, sink: str
+) -> str:
+    """
+    Builds a GStreamer command to process the golden sample.
+
+    :param gst_bin:
+        The binary name of gstreamer. Default is "gst-launch-1.0"
+        You can assign the snap name to GST_LAUNCH_BIN env variable if you
+        want to using snap.
+    :param golden_sample:
+        The path to the golden sample file.
+    :param decoder:
+        The decoder to use for the video, e.g., "v4l2vp8dec", "v4l2vp9dec".
+    :param sink:
+        The desired sink option, e.g., "fakesink".
+
+    :returns:
+        The GStreamer command to execute.
+    """
+    cmd = (
+        "{} -v filesrc location={} ! parsebin ! queue ! {} ! queue ! "
+        "v4l2convert output-io-mode=dmabuf-import capture-io-mode=dmabuf ! "
+        'queue ! fpsdisplaysink video-sink="{}"'
+        " text-overlay=false sync=false"
+    ).format(gst_bin, golden_sample_path, decoder, sink)
+
+    return cmd
+
+
+def execute_command(cmd: str) -> str:
+    """
+    Executes the GStreamer command and extracts the specific data from the
+    output. The specific data is the value of last-message which is exposed by
+    fpsdisplaysink.
+
+    :param cmd:
+        The GStreamer command to execute.
+
+    :returns:
+        The extracted last_message.
+    """
+    try:
+        logging.info("Starting command: '{}'".format(cmd))
+        ret = subprocess.run(
+            shlex.split(cmd),
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            encoding="utf-8",
+            timeout=30,
+        )
+        logging.info(ret.stdout)
+        return ret.stdout
+    except Exception as e:
```

Comment:
```py
except subprocess.CalledProcessError as e
```

Reply:
Good catch, thanks.

Reply:
What do you think if I change it as below ? Log the error seems like not necessary here.

```suggestion
    except Exception as e:
        raise SystemExit(e)
```

Reply:
:+1:

Reply:
Solved